src/OrchardCore.Modules/OrchardCore.Deployment/Controllers/ImportController.cs

```diff
@@ -148,6 +154,21 @@ public async Task<IActionResult> Json(ImportJsonViewModel model)
 
                     await _notifier.SuccessAsync(H["Recipe imported."]);
                 }
+                catch (RecipeExecutionException e)
+                {
+                    _logger.LogError(e, "Unable to import a recipe from JSON input.");
+
+                    foreach (var entry in e.StepResult.Errors)
+                    {
+                        ModelState.AddModelError(nameof(model.Json), string.Join(' ', entry.Value));
```

Comment:
Does this error show up on the page properly?

Reply:
Yes. It worked for me. I tested it.

Reply:
Yes because the view has `@Html.ValidationSummary()` but when importing from a package it doesn't show the error. Importing from packages doesn't seem to be handled in this PR

Reply:
In addition to the package import functionality, there is also remote deployment `ImportRemoteInstanceController` that requires the same treatment

Reply:
I recommend using Notify because it is automatic and we don't need to make adjustments to the view.

Reply:
@hyzx86 I am using ModelState when we are dealing with form otherwise notifications. I added screenshots of the 3 ways recipes are imported/run.